orchestrator/src/compression/blob.rs

```diff
@@ -0,0 +1,294 @@
+        }
+    }
+
+    // Check for any addresses with only nonce updates that weren't processed
+    for (address, nonce) in nonces.iter() {
+        if !processed_addresses.contains(address) {
+            leftover_addresses.push((*address, None, Some(*nonce)));
+            tracing::debug!("Found leftover nonce: address={}, nonce={}", address, nonce);
+        }
+    }
+
+    // Sort leftover addresses for deterministic output
+    leftover_addresses.sort_by_key(|(address, _, _)| *address);
+
+    tracing::debug!(
+        "Processing {} leftover addresses with nonce or class updates but no storage updates",
+        leftover_addresses.len()
+    );
+
+    // Process each leftover address
+    for (address, class_hash, nonce) in leftover_addresses.clone() {
+        // Create DA word with zero storage entries
+        let da_word = da_word(class_hash.is_some(), nonce, 0, version)?;
+
+        tracing::debug!(
+            "Processing leftover address {}: class_hash={:?}, nonce={:?}",
+            address,
+            class_hash.map(|h| h.to_string()),
+            nonce.map(|n| n.to_string())
+        );
+
+        // Add address and DA word to blob data
+        blob_data.push(address);
+        blob_data.push(da_word);
+
+        // If there's a class hash, add it to blob data
+        if let Some(hash) = class_hash {
+            blob_data.push(hash);
+            tracing::debug!("Adding class hash {} for leftover address {}", hash, address);
+        }
+    }
+
+    // Update the first element with the total number of contracts (original storage diffs and leftover addresses)
+    let total_contracts = state_diff.storage_diffs.len() + leftover_addresses.len();
+    blob_data[0] = Felt::from(total_contracts);
+    tracing::debug!("Total contract updates in blob: {}", total_contracts);
+
+    // Add declared classes count
+    blob_data.push(Felt::from(state_diff.declared_classes.len()));
+
+    // Sort declared classes by class_hash for deterministic output
+    state_diff.declared_classes.sort_by_key(|class| class.class_hash);
+
+    // Process each declared class
+    for DeclaredClassItem { class_hash, compiled_class_hash } in state_diff.declared_classes.into_iter() {
+        blob_data.push(class_hash);
+        blob_data.push(compiled_class_hash);
+    }
+
+    tracing::debug!("Created blob data with {} elements", blob_data.len());
+
+    Ok(blob_data)
+}
+
+/// Creates a DA word with information about a contract
+///
+/// # Arguments
+/// * `class_flag` - Indicates if a new class hash is present
+/// * `nonce_change` 0 Optional nonce value as a Felt
+/// * `num_change` - Number of storage updates
+/// * `version` - Version string to determine the encoding format
+///
+/// # Returns
+/// A `Felt` representing the encoded DA word
+pub fn da_word(
+    class_flag: bool,
+    nonce_change: Option<Felt>,
+    num_changes: u64,
+    version: StarknetVersion,
+) -> Result<Felt, JobError> {
```

Comment:
The starknet documentation uses U256, but a Felt does not necessarily contain a U256
```suggestion
) -> Result<U256, JobError> {
```

Reply:
da_word is part of the data we send for state update. It should always fit in a Felt